content/mooreslaw-tutorial.md

```diff
@@ -0,0 +1,579 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.6.0
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Determining Moore's Law with real data in NumPy
+![image.png](01-mooreslaw-tutorial-intro.png)
+
+_The number of transistors reported on a given chip plotted on a log scale in the y axis with the date of introduction on the linear scale x-axis. The blue data points are from a [transistor count table](https://en.wikipedia.org/wiki/Transistor_count#Microprocessors). The red line is an ordinary least squares prediction and the orange line is Moore's law._
+
+## What you'll do
+
+In 1965, engineer Gordon Moore
+[predicted](https://en.wikipedia.org/wiki/Moore%27s_law) that
+transistors on a chip would double every two years in the coming decade
+[[1](https://en.wikipedia.org/wiki/Moore%27s_law),
+[2](https://newsroom.intel.com/wp-content/uploads/sites/11/2018/05/moores-law-electronics.pdf)].
+You'll compare Moore's prediction against actual transistor counts in
+the 53 years following his prediction. You will determine the best-fit constants to describe the exponential growth of transistors on semiconductors compared to Moore's Law.
+
+
+## Skills you'll learn
+
+
+- Load data from a [\*.csv](https://en.wikipedia.org/wiki/Comma-separated_values) file
+- Perform linear regression and predict exponential growth using ordinary least squares
+- You'll compare exponential growth constants between models
+- Share your analysis in a file:
+    - as NumPy zipped files `*.npz` 
+    - as a `*.csv` file
+- Assess the amazing progress semiconductor manufacturers have made in the last five decades
+
+## What you'll need
+
+1. These packages:
+
+    * NumPy
+    * [Matplotlib](https://matplotlib.org/)
+    * [statsmodels](https://www.statsmodels.org) ordinary linear regression
+    
+    imported with the following commands
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+import numpy as np
+import statsmodels.api as sm
+```
+
+2. Since this is an exponential growth law you need a little background in doing math with [natural logs](https://en.wikipedia.org/wiki/Natural_logarithm) and [exponentials](https://en.wikipedia.org/wiki/Exponential_function). 
```

Comment:
@melissawm, there is a `region` defined that ends after item 1 e.g. 
````
<!-- #region --> 
# My list
1. item 1
```python 
import numpy as np
```
<!-- #endregion -->
2. item 2
````

I'm not sure that would cause the numbering reset, I think its a cell-level problem. I think there are two workarounds:
1. code in the numbers as you did either `$1.$` or __1.__, but the list is no longer indented, it just starts the line with the 1
2. remove the code-cell and use a 4-space indent for the block of code as such
    
````
1. item 1
    ```python
    import numpy as np
    ```
2. item 2
````

I haven't been able to get a another method to work within Jupyter's common markdown

Reply:
I like the second option, but I'm concerned this might impact a notebook automatically generated from the md file - if the cell is not marked as a code-cell, jupytext probably won't translate it to a code cell, right?

Reply:
It will have to be included a second time as an actual code cell, but we could add a tag and hide the cell from the html build. When viewing in a notebook, it would appear twice. Maybe we could put it in the first cell.

Reply:
I think the issue is that the code cell in between the numbers breaks up the highlighted chunk into three distinct cells: markdown-code-markdown. Numbering within a markdown cell is handled per the commonmark standard resetting the numbering per cell.

The simplest way to get around the markdown numbering is e.g. bold the numbers:

```suggestion
**1.** These packages:

* NumPy
* [Matplotlib](https://matplotlib.org/)
* [statsmodels](https://www.statsmodels.org) ordinary linear regression
    
imported with the following commands

```{code-cell} ipython3
import matplotlib.pyplot as plt
import numpy as np
import statsmodels.api as sm
```

**2.** Since this is an exponential growth law you need a little background in doing math with [natural logs](https://en.wikipedia.org/wiki/Natural_logarithm) and [exponentials](https://en.wikipedia.org/wiki/Exponential_function). 
```

Reply:
Oh I didn't realize that - this is the simplest and cleanest solution. Thanks!